src/Stratis.Bitcoin.Features.MemoryPool/MemPoolCoinView.cs

```diff
@@ -86,7 +86,7 @@ public override Task<uint256> Rewind()
                 return this.memPool.MapTx.Values.Where(t => ids.Contains(t.TransactionHash)).Select(s => s.Transaction).ToList();
             });
             IEnumerable<UnspentOutputs> memOutputs = mempoolcoins.Select(s => new UnspentOutputs(TxMempool.MempoolHeight, s));
-            coins.UnspentOutputs = coins.UnspentOutputs.Concat(memOutputs).ToArray();
+            coins = new FetchCoinsResponse(coins.UnspentOutputs.Concat(memOutputs).ToArray(), coins.BlockHash);
```

Comment:
You think this is better? well why not.

Reply:
it prevents null assignment to that member